Code refactoring

furkanmtorun opened the pull request:
Hi Max @straussmaximilian,

Here is the code refactoring PR that is referred in the issue #4 .
I tried:
-  to make the functions shorter 
- to make the functions decoupled component-wise.
- to bring all UI texts to one file so that easier to fix.
- to give better file/functions orchestration
- to make main `omiclearn.py` file clean.

I hope this makes the code easier to read.

Could you please also run the app and test it out?
I look forward to hearing from you!

Cheers

Commit:
REFACTOR: Extract all UI texts and bring them together

Commit:
REFACTOR: checkpoint_for_data_upload func

Commit:
REFACTOR: bulky classify_and_plot()

Commit:
REFACTOR: OmicLearn_Main func

Commit:
update: black formatting

straussmaximilian reviewed (approved):
Overall, looks good. I like the idea to keep ui components somewhere else. Some suggestions:

- How about `ui_components` instead of `ui_helper`?
- The usage of an underscore `_` seems not consistent to me, e.g.  in `_generate_imputation_elements` it is a standalone function, in `generate_sidebar_elements` it is used for internal functions. Would it make sense to use OOP here so that it is clear which function belongs to what.
- The text import is a bit redundant. `ui_texts` import `..._TEXT`.

Commit:
RENAME to ui_components and FIX text imports

Commit:
UPDATE: Func naming and add more comments

furkanmtorun commented:
Thanks for the valuable suggestions Max!

- I renamed to `ui_components.`
- Fixed the text imports : `from omiclearn.utils.ui_texts import *`
- As for use of underscore `_`, I tried to have a convention. I removed the `_` from all standalone functions (like all plotting functions, all functions in `ml_helper`) while I keep the `_` for internal functions (the ones that are called within those standalone ones). For instance, I named `_generate_eda_section` since it is called within `dataset_handling`.
- OOP is also a good idea but it is also difficult to place the functions to one class since here some might belong to two parts. 

Thanks for your feedback! 
See you in the next PR :)
